### ISSUE TITLE
Custom resource deployment hangs forever

### ISSUE THREAD

**Issue by sean-legitscript** [serverless/serverless#12804]
### Issue description

We have a lambda that uses S3 events to be triggered.
```
processSpreadsheet:
  architecture: arm64
  memorySize: 3008
  timeout: 900
  handler: src/_handlers.processSpreadsheet
  events:
    - s3:
        bucket: example-bucket
        event: s3:ObjectCreated:*
        existing: true
        forceDeploy: true
```

Our team needs to deploy this lambda for each of our stages. However, after we successfully deploy this lambda for one stage, all other sls deployments hang. After about an hour, the deployment fails with:
```
✔ Service packaged (89s)
(node:31517) NOTE: We are formalizing our plans to enter AWS SDK for JavaScript (v2) into maintenance mode in 2023.

Please migrate your code to use AWS SDK for JavaScript (v3).
For more information, check the migration guide at https://a.co/7PzMCcy
(Use `node --trace-warnings ...` to show where the warning was created)

Deploying merchantsmyls to stage qa (us-west-2)

✖ Stack merchantsmyls-qa failed to deploy (3777s)
Environment: darwin, node 20.11.1, framework 3.38.0 (local), plugin 7.2.3, SDK 4.5.1
Credentials: Local, "default" profile
Docs:        docs.serverless.com
Support:     forum.serverless.com
Bugs:        github.com/serverless/serverless/issues

Error:
UPDATE_FAILED: ProcessSpreadsheetCustomS31 (Custom::S3)
CloudFormation did not receive a response from your Custom Resource. Please check your logs for requestId [3c8d63e8-087d-48fc-938b-87e266eca2c4]. If you are using the Python cfn-response module, you may need to update your Lambda function code so that CloudFormation can attach the updated version.
```

We need to be able to deploy this lambda for more then one stage. We thought that using the forceDeploy: true flag would allow us to push forward but it still doesn't work. We need a way to push forward.

### Context

Serverless Version: 3.38.0

Plugins:
- serverless-api-gateway-caching
- serverless-v2-aws-documentation
- serverless-domain-manager
- serverless-offline
- serverless-step-functions
- serverless-plugin-aws-alerts
- serverless-webpack
- serverless-dependson-plugin

**Comment by sean-legitscript** [comment]
FYI, I realize this is a follow-up to my [previously closed ticket](https://github.com/serverless/serverless/issues/12597). However, I don't think I was given a sufficient answer as to why this is happening and didn't have time to respond before the ticket was closed.

**Comment by skierkowski** [comment]
@sean-legitscript have you tried to reproduce this issue with v4? If you try with V4 and get the error, then run `serverless support --github` to get more details you can share with us.

**Comment by sean-legitscript** [comment]
Hey gang, thanks for reaching out. However, I am looking for a better answer then bumping the version and asking ChatGPT. I want to get to the root cause of the issue but don't seem to have the resources to do that.

**Comment by sean-legitscript** [comment]
When I check the cloudwatch logs for the `custom-resource-existing-s3` lambda, I get the error:
```
{
    "errorType": "Runtime.ImportModuleError",
    "errorMessage": "Error: Cannot find module 'aws-sdk'\nRequire stack:\n- /var/task/utils.js\n- /var/task/s3/lib/permissions.js\n- /var/task/s3/handler.js\n- /var/runtime/index.mjs",
    "stack": [
        "Runtime.ImportModuleError: Error: Cannot find module 'aws-sdk'",
        "Require stack:",
        "- /var/task/utils.js",
        "- /var/task/s3/lib/permissions.js",
        "- /var/task/s3/handler.js",
        "- /var/runtime/index.mjs",
        "    at _loadUserApp (file:///var/runtime/index.mjs:1087:17)",
        "    at async UserFunction.js.module.exports.load (file:///var/runtime/index.mjs:1119:21)",
        "    at async start (file:///var/runtime/index.mjs:1282:23)",
        "    at async file:///var/runtime/index.mjs:1288:1"
    ]
}
```

**Comment by sean-legitscript** [comment]
Was able to successfully deploy with the help of [this comment](https://github.com/serverless/serverless/issues/11337#issuecomment-2176627178). Bumping from 3.38 to 3.39 solved the problem!

**Comment by skierkowski** [comment]
@sean-legitscript glad you got this resolved with the help of the community.

I'm going to close this issue out now that it is resolved (both in 3.39 and 4.0+).
